src/Sidekick.Business/Whispers/WhisperService.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Sidekick.Core.Initialization;
+using Sidekick.Core.Natives;
+
+namespace Sidekick.Business.Whispers
+{   
+    public class WhisperService : IWhisperService, IOnAfterInit
+    {
+        private INativeProcess pathOfExileProcess;
+        private string ClientLogFile;
+        public WhisperService(INativeProcess pathOfExileProcess)
+        {
+            this.pathOfExileProcess = pathOfExileProcess;
+        }
+
+        public string GetLatestWhisperCharacterName()
+        {
+            if (!pathOfExileProcess.IsPathOfExileInFocus || ClientLogFile == null) return String.Empty;
+
+            using (var stream = new FileStream(this.ClientLogFile, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+            {
+                if (stream.Length == 0)
+                {
+                    return null;
+                }
+                stream.Position = stream.Length - 1;
+                while(stream.Position > 0)
+                {
+                    var currentLine = GetLine(stream);
+
+                    // See if the current line contains a received whisper
+                    var match = Regex.Match(currentLine, @"(@From){1}\s.+?(?=:)", RegexOptions.Singleline);
+                    if(match.Success)
+                    {
+                        // No extract only character name
+                        int lastWhitespacePos = match.Value.LastIndexOf(" ") + 1;
+                        return match.Value.Substring(lastWhitespacePos, match.Value.Length - lastWhitespacePos);
+                    }
+
+                    stream.Position = stream.Position - 1;
+                }               
+            }
+            return null;
+        }
+
+        public Task OnAfterInit()
+        {
+            this.ClientLogFile = pathOfExileProcess.ClientLogPath;
+            return Task.CompletedTask;
```

Comment:
I would like not having to track the ClientLogFile locally inside this service. Is it really heavy to call the pathOfExileProcess.ClientLogPath every time?

Reply:
Does this work when we start Sidekick while Path of Exile is closed?

Reply:
You're absolutely right, this would not work when Path of Exile is closed.
I'll also change the code, so pathOfExileProcess.ClientLogPath will be called inside the method it is used and I'll delete the IOnAfterInit interface.

Reply:
Just pushed the changes